df2gspread/df2gspread.py

```diff
@@ -102,12 +102,18 @@ def upload(df, gfile="/New Spreadsheet", wks_name=None, chunk_size=1000,
         last_col = ascii_uppercase[seq_num % len(ascii_uppercase)] + last_col
         seq_num = seq_num // len(ascii_uppercase) - 1
 
-    # If user requested to resize sheet to fit dataframe or dataframe is larger than sheet,
-    # go ahead and resize it to better match new size of pandas dataframe.
-    # Otherwise, leave it the same size.
-    if df_size or len(df.index) + 1 > wks.row_count or len(df.columns) + 1 > wks.col_count:
+    # If user requested to resize sheet to fit dataframe, go ahead and 
+    # resize larger or smaller to better match new size of pandas dataframe.
+    # Otherwise, leave it the same size unless the sheet needs to be expanded
+    # to accomodate a larger dataframe.
+    if len(df.index) + 1 > wks.row_count:
+        wks.add_rows(len(df.index) - wks.row_count + 1)
+    if len(df.columns) + 1 > wks.col_count:
+        wks.add_cols(len(df.columns) - wks.col_count + 1)
+    if df_size:
```

Comment:
Yes, makes sense. I think I put it after because I was dealing with a bug where if the sheet was full (same amount of data as cells), the clean function would add extra rows and cols and I compensated by resizing after. But I think I fixed the bug in this latest commit by only adding extra rows and cols if needed.